src/unity/python/turicreate/test/test_one_shot_object_detector.py

```diff
@@ -0,0 +1,233 @@
+        if image_format == 'raw':
+            image = np.array(pil_img)
+            FORMAT_RAW = 2
+            return tc.Image(_image_data=image.tobytes(),
+                            _width=image.shape[1],
+                            _height=image.shape[0],
+                            _channels=image.shape[2],
+                            _format_enum=FORMAT_RAW,
+                            _image_data_size=image.size)
+        else:
+            with tempfile.NamedTemporaryFile(mode='w+b', suffix='.' + image_format) as f:
+                pil_img.save(f, format=image_format)
+                return tc.Image(f.name)
+
+    num_examples = 100
+    num_starter_images = 5
+    max_num_boxes_per_image = 10
+    classes = _CLASSES
+    images = []
+    FORMATS = ['png', 'jpeg', 'raw']
+    for i in range(num_examples):
+        # Randomly determine image size (should handle large and small)
+        img_shape = tuple(rs.randint(100, 1000, size=2)) + (3,)
+        img = rs.randint(255, size=img_shape)
+
+        pil_img = _PIL_Image.fromarray(img, mode='RGB')
+        # Randomly select image format
+        image_format = FORMATS[rs.randint(len(FORMATS))]
+        images.append(from_pil_image(pil_img, image_format=image_format))
+
+        
+
+    starter_images = []
+    starter_target = []
+    for i in range(num_starter_images):
+        img_shape = tuple(rs.randint(100, 1000, size=2)) + (3,)
+        img = rs.randint(255, size=img_shape)
+        pil_img = _PIL_Image.fromarray(img, mode='RGB')
+        image_format = FORMATS[rs.randint(len(FORMATS))]
+        starter_images.append(from_pil_image(pil_img, image_format=image_format))
+        starter_target.append(i % len(_CLASSES))
+
+    train = tc.SFrame({
+        feature: tc.SArray(starter_images),
+        target: tc.SArray(starter_target),
+    })
+    test = tc.SFrame({
+        feature: tc.SArray(images),
+    })
+    backgrounds = test[feature].head(5)
+    return train, test, backgrounds
+
+
+class OneObjectDetectorSmokeTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        """
+        The setup class method for the basic test case with all default values.
+        """
+        self.feature = 'myimage'
+        self.target = 'mytarget'
+
+        ## Create the model
+        self.def_opts = {
+           'model': 'darknet-yolo',
+           'max_iterations': 2,
+        }
+
+        # Model
+        self.train, self.test, self.backgrounds = _get_data(feature=self.feature, target=self.target)
+        self.model = tc.one_shot_object_detector.create(self.train,
+                                                        target=self.target,
+                                                        backgrounds=self.backgrounds,
+                                                        batch_size=2,
+                                                        max_iterations=1)
+
+        ## Answers
+        self.opts = self.def_opts.copy()
```

Comment:
It looks like `self.opts` is never getting used. . . . which means you don't need this line or the next line or the declaration of `self.def_opts`.